src/test/java/io/cdap/plugin/gcp/bigquery/relational/BigQueryRelationTest.java

```diff
@@ -577,6 +655,169 @@ public void testCollectDeduplicateAggregationDefinitionErrors() {
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+
+    // Check invalid select field
+    selectFields.put("a", new InvalidSQLExpression("a"));
+    partitionFields.add(factory.compile("a"));
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(factory.compile("a"),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+    // Check unsupported Select field
+    selectFields.put("a", new NonSQLExpression());
+    partitionFields.add(factory.compile("a"));
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(factory.compile("a"),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+    // Check invalid partition field
+    selectFields.put("a", factory.compile("a"));
+    partitionFields.add(new InvalidSQLExpression("a"));
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(factory.compile("a"),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+    // Check unsupported partition field
+    selectFields.put("a", factory.compile("a"));
+    partitionFields.add(new NonSQLExpression());
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(factory.compile("a"),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+    // Check invalid order field
+    selectFields.put("a", factory.compile("a"));
+    partitionFields.add(factory.compile("a"));
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(new InvalidSQLExpression("a"),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+
+    // Check unsupported order field
+    selectFields.put("a", factory.compile("a"));
+    partitionFields.add(factory.compile("a"));
+    orderByExpressions.add(new WindowAggregationDefinition.OrderByExpression(new NonSQLExpression(),
+      WindowAggregationDefinition.OrderBy.ASCENDING));
+    Assert.assertFalse(BigQueryRelation.supportsWindowAggregationDefinition(def));
+    selectFields.clear();
+    partitionFields.clear();
+    orderByExpressions.clear();
+  }
+
+  @Test
+  public void testCollectWindowAggregationDefinitionErrors() {
```

Comment:
From what I can see there are multiple test scenarios being tested in a single test here. Can we divide these into different methods so that we can see which test breaks at a glance?

Reply:
Followed the pattern of existing unit tests for other pushdowns. We are asserting for every field.So it is easy to identify at what place the test breaks.